File: NutriSmart/app/src/main/java/com/example/nutrismart/data/repository/MealRepository.kt
package com.example.nutrismart.data.repository

import com.example.nutrismart.data.remote.MealDto
import com.example.nutrismart.data.remote.MealPlanDto
import com.example.nutrismart.data.remote.NutriSmartApi
import com.example.nutrismart.data.remote.ShoppingListDto
import retrofit2.Response
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class MealRepository @Inject constructor(
    private val api: NutriSmartApi
) {
    suspend fun getDailyPlan(token: String, userId: Long, date: String): Response<MealPlanDto> {
        return api.getDailyPlan("Bearer $token", userId, date)
    }

    suspend fun getShoppingList(token: String, userId: Long): Response<ShoppingListDto> {
        return api.getShoppingList("Bearer $token", userId)
    }

    suspend fun getMealAlternatives(token: String, userId: Long, type: String): Response<List<MealDto>> {
        return api.getMealAlternatives("Bearer $token", userId, type)
    }

    suspend fun swapMeal(token: String, userId: Long, type: String, mealId: Long, date: String): Response<MealPlanDto> {
        return api.swapMeal("Bearer $token", userId, type, mealId, date)
    }

    suspend fun toggleMealConsumed(token: String, mealId: Long, consumed: Boolean): Response<MealDto> {
        return api.toggleMealConsumed("Bearer $token", mealId, consumed)
    }

    suspend fun toggleShoppingItem(token: String, itemId: Long, checked: Boolean): Response<Unit> {
        return api.toggleShoppingItem("Bearer $token", itemId, checked)
    }
}

File: NutriSmartIOS/data/repository/MealRepository.swift
//
//  MealRepository.swift
//  NutriSmartIOS
//

import Foundation

import Foundation

class MealRepository {
    static let shared = MealRepository()
    
    // Conectăm repository-ul la serviciul nostru de rețea
    private let api = NutriSmartApiService.shared
    
    private init() {}
    
    func getDailyPlan(token: String, userId: Int, date: String) async throws -> MealPlanDto {
        return try await api.getDailyPlan(token: token, userId: userId, date: date)
    }

    func getShoppingList(token: String, userId: Int) async throws -> ShoppingListDto {
        return try await api.getShoppingList(token: token, userId: userId)
    }

    func getMealAlternatives(token: String, userId: Int, type: String) async throws -> [MealDto] {
        return try await api.getMealAlternatives(token: token, userId: userId, mealType: type)
    }

    func swapMeal(token: String, userId: Int, type: String, mealId: Int, date: String) async throws -> MealPlanDto {
        return try await api.swapMeal(token: token, userId: userId, mealType: type, newMealId: mealId, date: date)
    }

    func toggleMealConsumed(token: String, mealId: Int, consumed: Bool) async throws -> MealDto {
        return try await api.toggleMealConsumed(token: token, mealId: mealId, consumed: consumed)
    }

    func toggleShoppingItem(token: String, itemId: Int, checked: Bool) async throws -> EmptyResponse {
        return try await api.toggleShoppingItem(token: token, itemId: itemId, isChecked: checked)
    }
}
